dia-batching-server/resources/schema.graphql

```diff
@@ -0,0 +1,894 @@
+}
+
+type StableSwapLiquidityPosition {
+  id: ID!
+  user: User!
+  stableSwap: StableSwap!
+  liquidityTokenBalance: String!
+}
+
+# saved over time for return calculations, gets created and never updated
+type LiquidityPositionSnapshot {
+  id: ID!
+  liquidityPosition: LiquidityPosition!
+  timestamp: DateTime! # saved for fast historical lookups
+  block: Int! # saved for fast historical lookups
+  user: User! # reference to user
+  pair: Pair! # reference to pair
+  "BigDecimal"
+  token0PriceUSD: String! # snapshot of token0 price
+  "BigDecimal"
+  token1PriceUSD: String! # snapshot of token1 price
+  "BigDecimal"
+  reserve0: String! # snapshot of pair token0 reserves
+  "BigDecimal"
+  reserve1: String! # snapshot of pair token1 reserves
+  "BigDecimal"
+  reserveUSD: String! # snapshot of pair reserves in USD
+  "BigDecimal"
+  liquidityTokenTotalSupply: String! # snapshot of pool token supply
+  "BigDecimal"
+  liquidityTokenBalance: String! # snapshot of users pool token balance
+}
+
+type Transaction {
+  id: ID! # txn hash
+  blockNumber: BigInt!
+  timestamp: DateTime!
+  mints: [ID!]!
+  burns: [ID!]!
+  swaps: [ID!]!
+}
+
+type Mint {
+  # transaction hash + "-" + index in mints Transaction array
+  id: ID!
+  transaction: Transaction!
+  timestamp: DateTime! # need this to pull recent txns for specific token or pair
+  pair: Pair!
+
+  # populated from the primary Transfer event
+  to: String!
+  liquidity: String!
+
+  # populated from the Mint event
+  sender: String
+  amount0: String
+  amount1: String
+  logIndex: Int
+  # derived amount based on available prices of tokens
+  amountUSD: String
+
+  # optional fee fields, if a Transfer event is fired in _mintFee
+  feeTo: String
+  feeLiquidity: String
+}
+
+type Burn {
+  # transaction hash + "-" + index in mints Transaction array
+  id: ID!
+  transaction: Transaction!
+  timestamp: DateTime! # need this to pull recent txns for specific token or pair
+  pair: Pair!
+
+  # populated from the primary Transfer event
+  liquidity: String!
+
+  # populated from the Burn event
+  sender: String
+  amount0: String
+  amount1: String
+  to: String
+  logIndex: Int
+  # derived amount based on available prices of tokens
+  amountUSD: String
+
+  # mark uncomplete in ETH case
+  needsComplete: Boolean!
+
+  # optional fee fields, if a Transfer event is fired in _mintFee
+  feeTo: String
+  feeLiquidity: String
+}
+
+type Swap {
+  # transaction hash + "-" + index in swaps Transaction array
+  id: ID!
+  transaction: Transaction!
+  timestamp: DateTime! # need this to pull recent txns for specific token or pair
+  pair: Pair!
+
+  # populated from the Swap event
+  sender: String!
+  from: String! # the EOA that initiated the txn
+  amount0In: String!
+  amount1In: String!
+  amount0Out: String!
+  amount1Out: String!
+  to: String!
+  logIndex: Int
+
+  # derived info
+  amountUSD: String!
+}
+
+type Bundle {
+  id: ID!
+  "BigDecimal"
+  ethPrice: Decimal! # price of ETH usd
+}
```

Comment:
We can probably limit the schema to just this type and remove all others right? Let's try to do that as the others will not need to be used in the batching server.